Applications/DataExplorer/DataView/DiagramView/DiagramPrefsDialog.cpp

```diff
@@ -67,47 +67,52 @@ DiagramPrefsDialog::~DiagramPrefsDialog()
 
 void DiagramPrefsDialog::accept()
 {
-    if ((fromDateLine->text().length() > 0) && (toDateLine->text().length() > 0) &&
-        (!_list.empty()))
+    QDateTime start_date(DiagramList::getDateTime(fromDateLine->text()));
+    QDateTime end_date(DiagramList::getDateTime(toDateLine->text()));
+
+    if (start_date == QDateTime() || end_date == QDateTime() ||
+        start_date > end_date || _list.empty())
     {
-        // data has been loaded
-        if (_list[0]->size() > 0)
-        {
-            bool window_is_empty(false);
-            if (_window == nullptr)
-            {
-                _window = new DetailWindow();
-                _window->setAttribute(Qt::WA_DeleteOnClose);
-                window_is_empty = true;
-            }
+        OGSError::box("No data found...");
+        return;
+    }
 
-            for (std::size_t i = 0; i < _list.size(); i++)
-                if (this->_visability[i]->isChecked())
-                {
-                    _window->addList(_list[i]);
-                    window_is_empty = false;
-                }
+    // data has been loaded
+    if (_list[0]->size() > 0)
+    {
+        bool window_is_empty(false);
+        if (_window == nullptr)
+        {
+            _window = new DetailWindow();
+            _window->setAttribute(Qt::WA_DeleteOnClose);
+            window_is_empty = true;
+        }
 
-            if (!window_is_empty)
-            {
-                _window->show();
-                this->done(QDialog::Accepted);
-            }
-            else
+        for (std::size_t i = 0; i < _list.size(); i++)
+            if (_visability[i]->isChecked())
```

Comment:
In line 81 it is checked the size for the first item of the data container. Why isn't it checked for the other items in the `_list`.

Reply:
I've added a comment above to make clear what the intent is here...